tests/unit/test_agw_network_configurator.py

```diff
@@ -143,6 +179,66 @@ def test_given_interfaces_dont_have_correct_names_when_update_interfaces_names_t
                         "match": {"macaddress": "02:45:b8:e6:23:c6"},
                         "set-name": "eth0",
                     },
+                    "eth1": {
+                        "dhcp4": True,
+                        "dhcp6": False,
+                        "match": {"macaddress": "02:45:b8:e6:c6:23"},
+                        "set-name": "eth1",
+                    },
+                },
+                "version": 2,
+            }
+        }
+
+        agw_network_configurator.update_interfaces_names()
+
+        args, kwargs = mock_yaml_dump.call_args
+        self.assertEqual(args[0], expected_cloud_init_content)
+        self.assertEqual(mock_yaml_dump.call_count, 1)
+
+    @patch("magma_access_gateway_installer.agw_network_configurator.yaml.dump")
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.yaml.safe_load",
+        return_value=copy.deepcopy(CLOUD_INIT_CONTENT_MORE_INTERFACES),
+    )
+    @patch("magma_access_gateway_installer.agw_network_configurator.open", new_callable=mock_open)
+    @patch("magma_access_gateway_installer.agw_network_configurator.check_call")
+    def test_given_custom_sgi_and_s1_interfaces_names_when_update_interfaces_names_then_netplan_config_file_is_updated(  # noqa: E501
+        self, _, __, ___, mock_yaml_dump
+    ):
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.INCORRECT_NETWORK_INTERFACES,
+            self.TEST_DNS_LIST,
+            sgi_interface=self.TEST_SGi_INTERFACE,
+            s1_interface=self.TEST_S1_INTERFACE,
+        )
+        expected_cloud_init_content = {
+            "network": {
+                "ethernets": {
+                    "test_if_name": {
+                        "dhcp4": True,
+                        "dhcp6": False,
+                        "match": {"macaddress": "02:45:b8:e6:23:c6"},
```

Comment:
Could we have variables for interfaces. Here it seems like we expect that the mac address is `02:45:b8:e6:23:c6` while reality is that we expect that mac address is whatever was initially there.

Reply:
Agreed. Refactored.